src/Avalonia.Controls/SelectionModel.cs

```diff
@@ -0,0 +1,860 @@
+            {
+                ClearSelection(resetAnchor: true);
+            }
+
+            var childNode = _rootNode.GetAt(groupIndex, realizeChild: true);
+            var selected = childNode!.Select(itemIndex, select);
+
+            if (selected)
+            {
+                AnchorIndex = new IndexPath(groupIndex, itemIndex);
+            }
+        }
+
+        private void SelectWithPathImpl(IndexPath index, bool select)
+        {
+            bool selected = false;
+
+            if (_singleSelect)
+            {
+                ClearSelection(resetAnchor: true);
+            }
+
+            SelectionTreeHelper.TraverseIndexPath(
+                _rootNode,
+                index,
+                true,
+                (currentNode, path, depth, childIndex) =>
+                {
+                    if (depth == path.GetSize() - 1)
+                    {
+                        selected = currentNode.Select(childIndex, select);
+                    }
+                }
+            );
+
+            if (selected)
+            {
+                AnchorIndex = index;
+            }
+        }
+
+        private void SelectRangeFromAnchorImpl(int index, bool select)
+        {
+            int anchorIndex = 0;
+            var anchor = AnchorIndex;
+
+            if (anchor != null)
+            {
+                anchorIndex = anchor.GetAt(0);
+            }
+
+            _rootNode.SelectRange(new IndexRange(anchorIndex, index), select);
+        }
+
+        private void SelectRangeFromAnchorWithGroupImpl(int endGroupIndex, int endItemIndex, bool select)
+        {
+            var startGroupIndex = 0;
+            var startItemIndex = 0;
+            var anchorIndex = AnchorIndex;
+
+            if (anchorIndex != null)
+            {
+                startGroupIndex = anchorIndex.GetAt(0);
+                startItemIndex = anchorIndex.GetAt(1);
+            }
+
+            // Make sure start > end
+            if (startGroupIndex > endGroupIndex ||
+                (startGroupIndex == endGroupIndex && startItemIndex > endItemIndex))
+            {
+                int temp = startGroupIndex;
+                startGroupIndex = endGroupIndex;
+                endGroupIndex = temp;
+                temp = startItemIndex;
+                startItemIndex = endItemIndex;
+                endItemIndex = temp;
+            }
+
+            for (int groupIdx = startGroupIndex; groupIdx <= endGroupIndex; groupIdx++)
+            {
+                var groupNode = _rootNode.GetAt(groupIdx, realizeChild: true)!;
+                int startIndex = groupIdx == startGroupIndex ? startItemIndex : 0;
+                int endIndex = groupIdx == endGroupIndex ? endItemIndex : groupNode.DataCount - 1;
+                groupNode.SelectRange(new IndexRange(startIndex, endIndex), select);
+            }
+        }
+
+        private void SelectRangeImpl(IndexPath start, IndexPath end, bool select)
+        {
+            var winrtStart = start;
```

Comment:
winrt ?

Reply:
Yeah, comes from WinUI. I try to avoid renaming stuff when porting to make porting subsequent changes easier, however in this case it might be worth changing it because I agree, this really stands out. What do you think?

Reply:
Well, from what I understood is that our `SelectionModel` differs from the original anyway so I figured we might want to unify some names. 

But I have no problem with keeping `winrt` although it looks a bit weird in our context.

Reply:
I think I'd rather keep the naming from upstream to ease porting of subsequent changes.